examples/notebooks/tour-of-beam/reading-and-writing-data.ipynb

```diff
@@ -724,16 +740,21 @@
         "    self.batch_size = batch_size\n",
         "\n",
         "  @staticmethod\n",
+        "  def expand_pattern(pattern):\n",
+        "    for match_result in beam_fs.match([pattern])[0].metadata_list:\n",
+        "      yield match_result.path\n",
+        "\n",
+        "  @staticmethod\n",
         "  def write_file(lines, file_name_prefix, file_name_suffix):\n",
         "    file_name = f\"{file_name_prefix}-{uuid.uuid4().hex}{file_name_suffix}\"\n",
-        "    with open(file_name, 'w') as f:\n",
+        "    with beam_fs.create(file_name) as f:\n",
         "      for line in lines:\n",
-        "        f.write(f\"{line}\\n\")\n",
+        "        f.write(f\"{line}\\n\".encode('utf-8'))\n",
         "\n",
         "  def expand(self, pcollection):\n",
         "    # Remove existing files matching the output file_name pattern.\n",
-        "    for f in glob.glob(f\"{self.file_name_prefix}*{self.file_name_suffix}\"):\n",
-        "      os.remove(f)\n",
+        "    for path in self.expand_pattern(f\"{self.file_name_prefix}*{self.file_name_suffix}\"):\n",
```

Comment:
Similarly, `WriteToText` might be used to simplify things.

Reply:
Again, this was more to show how to create custom I/O transforms, but I'll definitely add a note pointing them to `WriteToText`.